json-format.md

```diff
@@ -134,42 +132,38 @@ become members of the JSON object, with the respective JSON object member name
 matching the attribute name, and the member's type and value being mapped using
 the [type system mapping](#22-type-system-mapping).
 
-### 3.1. Special Handling of the "data" Attribute
+The data payload, if present, MUST be mapped to a member named `data`,
+according to the [Data Payload](#31-data-payload) section. If no data payload
+is present, the envelope MUST NOT contain a `data` member.
 
-The mapping of the `Any`-typed `data` attribute follows the rules laid out in
-[Section 2.3.](#23-mapping-any-typed-attributes), with two additional rules:
+### 3.1. Data Payload
 
-First, if an implementation determines that the type of the `data` attribute is
-`Binary` or `String`, it MUST inspect the `datacontenttype` attribute to
-determine whether it is indicated that the data value contains JSON data.
+The mapping of the data payload to a [JSON value][json-value] depends on the
+`datacontenttype` attribute.
 
-If the `datacontenttype` value is either ["application/json"][rfc4627] or any
-media type with a [structured +json suffix][rfc6839], the implementation MUST
-translate the `data` attribute value into a [JSON value][json-value], and set
-the `data` attribute of the envelope JSON object to this JSON value.
-
-If the `datacontenttype` value does not follow the [structured +json
-suffix][rfc6839] but is known to use JSON encoding, the implementation MUST
-translate the `data` attribute value into a [JSON value][json-value], and set
-the `data` attribute of the envelope JSON object to this JSON value. Its typical
-examples are, but not limited to, `text/json`,
+If the `datacontenttype` is [`"application/json"`][rfc4627], any structured
+media type with a [structured `+json` suffix][rfc6839], any other media type
+that is known by the implementation to use JSON encoding, or if through some
+other means the implementation knows that the payload is JSON, the data payload
```

Comment:
Note the "some other means" here is intended to allow for programmatic APIs, where the payload might allow any type, or where a polymorphic type is used that allows JSON AST, to set the data to JSON, even if the user hasn't explicitly set the content type to a JSON content type.